File: Cpp/407-lowhighswap-cpp/src/low_high_swap.cc
#include "low_high_swap.h"
using namespace std;

string vec_to_string(vector<int> array)
{
    string s = "";
    for (int i = 0; i < array.size(); i++) {
        s += to_string(array[i]);
    }
    return s;
}

string low_high_swap(vector<int> array)
{
    int len = array.size();
    for (int i = 0, j = len - 1; i < j; i++, j--) {
        if (array[i] > array[j]) {
            swap(array[i], array[j]);
        }
    }
    return vec_to_string(array);
}

File: Java/122-301-lowhighswap/src/main/java/LowHighSwap.java
public class LowHighSwap {
    // 请实现swap方法
    public String swap(int[] array) {
        int left = 0;
        int right = array.length - 1;
        for (; left <= right; left++, right--) {
            if (array[left] > array[right]) {
                int tmp = array[left];
                array[left] = array[right];
                array[right] = tmp;
            }
        }
        return array2String(array);
    }

    // 此方法无需修改
    private String array2String(int[] array) {
        StringBuilder sb = new StringBuilder();
        for (int n : array) {
            sb.append(n);
        }
        return sb.toString();
    }
}
